modules/base/rendering/renderablenodeline.cpp

```diff
@@ -0,0 +1,307 @@
+#include <openspace/util/updatestructures.h>
+#include <ghoul/filesystem/filesystem.h>
+#include <ghoul/opengl/programobject.h>
+#include <openspace/interaction/navigationhandler.h>
+#include <openspace/interaction/orbitalnavigator.h>
+
+namespace {
+
+    constexpr const char* ProgramName = "NodeLineProgram";
+    constexpr const char* Root = "Root";
+
+    constexpr openspace::properties::Property::PropertyInfo StartNodeInfo = {
+        "StartNode",
+        "Start Node",
+        "The identifier of the node the line starts from. "
+    };
+
+    constexpr openspace::properties::Property::PropertyInfo EndNodeInfo = {
+        "EndNode",
+        "End Node",
+        "The identifier of the node the line ends at. "
+    };
+
+    constexpr openspace::properties::Property::PropertyInfo LineColorInfo = {
+        "Color",
+        "Color",
+        "This value determines the RGB color for the line."
+    };
+
+    constexpr openspace::properties::Property::PropertyInfo LineWidthInfo = {
+        "LineWidth",
+        "Line Width",
+        "This value specifies the line width."
+    };
+
+} // namespace
+
+namespace openspace {
+
+documentation::Documentation RenderableNodeLine::Documentation() {
+    using namespace documentation;
+    return {
+        "Renderable Node Line",
+        "base_renderable_renderablenodeline",
+        {
+            {
+                StartNodeInfo.identifier,
+                new StringVerifier,
+                Optional::No,
+                StartNodeInfo.description
+            },
+            {
+                EndNodeInfo.identifier,
+                new StringVerifier,
+                Optional::No,
+                EndNodeInfo.description
+            },
+            {
+                LineColorInfo.identifier,
+                new DoubleVector3Verifier,
+                Optional::Yes,
+                LineColorInfo.description
+            },
+            {
+                LineWidthInfo.identifier,
+                new DoubleVerifier,
+                Optional::Yes,
+                LineWidthInfo.description
+            }
+        }
+    };
+}
+
+RenderableNodeLine::RenderableNodeLine(const ghoul::Dictionary& dictionary)
+    : Renderable(dictionary)
+    , _lineColor(LineColorInfo, glm::vec3(1.f, 1.f, 1.f), glm::vec3(0.f), glm::vec3(1.f))
+    , _lineWidth(LineWidthInfo, 2.f, 1.f, 20.f)
+    , _start(StartNodeInfo, Root)
+    , _end(EndNodeInfo, Root)
+{
+    documentation::testSpecificationAndThrow(
+        Documentation(),
+        dictionary,
+        "RenderableNodeLine"
+    );
+
+    _start = dictionary.value<std::string>(StartNodeInfo.identifier);
+    _end = dictionary.value<std::string>(EndNodeInfo.identifier);
+
+    if (dictionary.hasKey(LineColorInfo.identifier)) {
+        _lineColor = dictionary.value<glm::vec3>(LineColorInfo.identifier);
+    }
+    if (dictionary.hasKey(LineWidthInfo.identifier)) {
+        _lineWidth = float(dictionary.value<double>(LineWidthInfo.identifier));
```

Comment:
Prefer C++ style casts over C casts:
`static_cast<float>(dictionary.value<double>(LineWidthInfo.identifier));`

Reasoning:  A C-cast can mean many different things and it requires some thinking to figure out what is meant in each situation.  The C++ style makes it more clear and prevent bugs from being introduced in general